File: frontend/src/types/game.ts
export interface GameState {
  loyalty: number
  doubt: number
  fatigue: number
}

export interface Choice {
  id: string
  text: string
  next_scene_id: string
  effects?: Partial<GameState>
}

export type SceneType = 'system' | 'dialogue' | 'conditional' | 'ending'

export interface Scene {
  id: string
  type: SceneType
  speaker?: string
  text?: string
  choices: Choice[]
}

export interface StartResponse {
  scene: Scene
  initial_state: GameState
}

export interface ChoiceResponse {
  next_scene_id: string
  scene: Scene
  new_state: GameState
}

File: backend/app/schemas/game.py
from typing import Optional
from pydantic import BaseModel


class Choice(BaseModel):
    id: str
    text: str
    next_scene_id: str
    effects: Optional[dict[str, int]] = None


class Scene(BaseModel):
    id: str
    type: str
    speaker: Optional[str] = None
    text: Optional[str] = None
    choices: list[Choice] = []


class GameState(BaseModel):
    loyalty: int = 0
    doubt: int = 0
    fatigue: int = 0


class StartResponse(BaseModel):
    scene: Scene
    initial_state: GameState


class ChoiceRequest(BaseModel):
    choice_id: str
    state: GameState


class ChoiceResponse(BaseModel):
    next_scene_id: str
    scene: Scene
    new_state: GameState
